app/lib/regular_expressions.rb

```diff
@@ -0,0 +1,4 @@
+class RegularExpressions
+  EMAIL_REGEXP =
+    %r{\A[a-zA-Z0-9.!\#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+\z}
```

Comment:
So what I don't like about this is that this regular expression is now entirely our problem. We're taking it on trust there's no typo in there. We have no tests to prove that it works, and no way of confirming that it's the same in most regards as a proven, battle-tested third party regex. It's also complex enough to be hard to just read and grok. So doing this means we just have to kind of trust and hope.

I can think of a couple of possible approaches that might make this less scary:

* Construct our regex by building it from `URI::MailTo::EMAIL_REGEXP` _plus_ some extra rules, and just adding tests for those rules
* Adding a boatload of tests of this regex, to demonstrate that all sane email formats are allowed and all insane formats aren't.
* Leaving the existing validation in place, and adding a separate validation that also checks for the presence of a `.` in the second half of the address, and just adding tests for this second validation

Reply:
I'm adding tests for it now yeah, we can't easily just construct it from the existing REGXP...

I like #3 the best, how would you pass multiple parameters into `with:` then?

Reply:
So you can't pass multiple regexes to `with`. You could possibly add a _second_ `validates :email` line, but from memory that leads to some slightly weird UI quirks. So I think you'd be best off adding a `validate :secondary_email_regex` line (note lack of `s` on validate), then defining something like:

```
def secondary_email_regex
  errors.add(:email, :invalid) unless /.*@.*\..*_or_whatever_/ =~ contact_email
end
```

Note you'd also want to do this in the NSM solicitor form

Reply:
Adjusted to only validate the TLD now